.github/workflows/build_workflow.yml

```diff
@@ -0,0 +1,262 @@
+    runs-on: ubuntu-latest
+    needs: [ checkstyle, spotbugs ]
+    steps:
+      -   name: Checkout Repository
+          uses: actions/checkout@v4
+
+      -   name: Setup Java 21 LTS
+          uses: actions/setup-java@v4
+          with:
+            java-version: 21
+            distribution: 'temurin'
+
+      -   name: Setup Gradle
+          uses: gradle/actions/setup-gradle@v4
+
+      -   name: Start Docker Dependencies
+          env:
+            GPC_CONSUMER_SERVER_PORT: "8090"
+            GPC_CONSUMER_SDS_URL: "http://wiremock:8080/spine-directory/"
+            GPC_CONSUMER_SDS_APIKEY: "anykey"
+            GPC_CONSUMER_LOGGING_LEVEL: DEBUG
+          run: |
+            docker network create commonforgpc
+            docker compose build
+            docker compose up gpc-consumer gpcc-mocks gpconnect-db gpconnect-api tkw gpcc-nginx --detach
+          working-directory: ./docker
+
+      -   name: Execute Integration Tests
+          run: ./gradlew integrationTest
+          working-directory: ./service
+
+      -   name: Dump Docker Logs
+          if: always()
+          run: |
+            chmod +x dump_docker_logs.sh
+            ./dump_docker_logs.sh
+          working-directory: ./scripts
+          shell: bash
+
+      -   name: Collect Artifacts
+          if: always()
+          run: |
+            mkdir -p artifacts
+            cp -r ./service/build/reports ./artifacts
+            cp -r ./scripts/logs ./artifacts
+
+      -   name: Upload Artifacts
+          uses: actions/upload-artifact@v4
+          if: always()
+          with:
+            name: 'Integration Test Reports & Docker Logs'
+            path: ./artifacts/**
+            compression-level: 9
+
+      -   name: Stop Docker Dependencies
+          if: always()
+          run: |
+            docker compose down --rmi=local --volumes --remove-orphans
+            docker compose rm
+            docker network rm commonforgpc
+          working-directory: ./docker
+
+      -   name: Temporary Artifacts Cleanup
+          run: rm -rf ./artifacts
+
+  build-and-publish-docker-images:
+      name: Build & Publish Docker Images
+      runs-on: ubuntu-latest
+      needs: [unit-tests, integration-tests]
+      permissions:
+          id-token: write
+          contents: read
+      strategy:
+          matrix:
+              config:
+                  - directory: service
+                    repository: gpc-consumer
+                    build-context: .
+
+      steps:
+          - name: Checkout Repository
+            uses: actions/checkout@v4
+
+          - name: Configure AWS Credentials
+            uses: aws-actions/configure-aws-credentials@v4
+            with:
+              role-to-assume: arn:aws:iam::${{ secrets.AWS_ACCOUNT_ID }}:role/${{ secrets.AWS_ROLE_TO_ASSUME }}
+              role-session-name: gpc_consumer_github_action_build_workflow
+              aws-region: ${{ secrets.AWS_REGION }}
+
+          - name: Generate Build ID
```

Comment:
@MartinWheelerMT has some comment github action which also posted the build ID as a comment on the PR which looked snazzy.